File: lernbegleiter-backend-app/src/main/java/at/technikumwien/lernbegleiter/data/dto/reflexion/WeeklyOverviewClass.java
package at.technikumwien.lernbegleiter.data.dto.reflexion;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.HashSet;
import java.util.Set;

@Data
@Accessors(chain = true)
@AllArgsConstructor
@NoArgsConstructor
public class WeeklyOverviewClass {
    private String uuid = "";
    private String name = "";
    private String color = "";
    private Set<WeeklyOverviewClassDay> days = new HashSet<>();
}

File: lernbegleiter-frontend-app/src/app/data/weekly-overview/WeeklyOverviewClass.ts
import { WeeklyOverviewClassDay } from './WeeklyOverviewClassDay';

export class WeeklyOverviewClass {
	uuid: string
	name: string
	color: string
	days: WeeklyOverviewClassDay[] = [new WeeklyOverviewClassDay(), new WeeklyOverviewClassDay(), new WeeklyOverviewClassDay(), new WeeklyOverviewClassDay(), new WeeklyOverviewClassDay()]
}